flux-middle/src/queries.rs

```diff
@@ -200,15 +200,18 @@ impl<'tcx> Queries<'tcx> {
 
     pub(crate) fn fn_sig(&self, genv: &GlobalEnv, def_id: DefId) -> QueryResult<rty::PolySig> {
         run_with_cache(&self.fn_sig, def_id, || {
-            if let Some(local_id) = def_id.as_local() {
+            // If it's an extern_fn, resolve it to its local fn_sig's def_id
+            let did =
+                if let Some(entry) = genv.extern_fns().get(&def_id) { *entry } else { def_id };
```

Comment:
maybe instead of exposing the entire `extern_fns` map we could only expose a function that tells you if the `DefId` is an extern spec

```rust
impl GlobalEnv {
   ...
   pub fn is_extern_spec(&self, def_id: DefId) -> Option<DefId> { ... }
 }
 ```
 Then  we could use it here like so
 
 ```rust
 let did = genv.is_extern_spec(def_id).unwrap_or_else(def_id);
```